### PR TITLE
Notification Message Updates

### PR DESCRIPTION
**Pull request by blakegreen32**
Updated NotificationMessage to new designs. Added description property that will add a description under the message that is optional when creating notification or calling this.notify(message, type, description);

Ticket: https://nanopay.atlassian.net/browse/NP-1010
nanopay pr: https://github.com/nanoPayinc/NANOPAY/pull/10044

### CODE REVIEW

**Comment by kgrgreer** [comment]
LGTM

File: src/foam/u2/dialog/NotificationMessage.js [hunk 1]
```diff
@@ -20,111 +20,204 @@ foam.CLASS({
...
+    }
+    ^close-icon:hover {
+      opacity: 1;
     }
   `,
 
+  properties: [
+    {
+      class: 'String',
+      name: 'type'
```
Comment (jlhughes) [inline src/foam/u2/dialog/NotificationMessage.js:140]: Can this use the same enum from foam.log.LogLevel?

Comment (blakegreen32) [reply src/foam/u2/dialog/NotificationMessage.js:140]: Its similar but I don't think so due to the fact that we need a success state, also the coloring isn't the same unless we plan to unify them. I'm up for the discussion though.

File: src/foam/u2/images/round-check-circle-green.svg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 20 20">
```
Comment (Nauna) [inline src/foam/u2/images/round-check-circle-green.svg:1]: so I've looked this up and actually found - we currently have 4 green check-marks - all the same ... :(
/foam/u2/images/checkmark-small-green.svg
/net/nanopay/images/checkmark-large-green.svg
/net/nanopay/images/checkmark-small-green.svg
/foam/nanos/controller/checkmark-small-green.svg

image

Comment (blakegreen32) [reply src/foam/u2/images/round-check-circle-green.svg:1]: haha oh wow, ok i'll remove the others thanks for checking that

File: src/foam/u2/dialog/NotificationMessage.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/u2/dialog/NotificationMessage.js:140]: re: colours - I don't really care what shade of green log info messages are. 
re: success - for notification? I don't see this anywhere - and  clearly shows the issue with free form text for state.  How would anyone to know what the 'type's are supposed to be.  I see notifications states identical to log messages: info, warning, error.  Where/how is success used?   @nanoscott tagging you here to comment as well if possible. Thanks.

Comment (nanoscott) [reply src/foam/u2/dialog/NotificationMessage.js:140]: info and success are basically the same .. scenarios where this is something we would want to display:
- Record was successfully created
- Approval request was approved
- etc.
All of these things are fine being shown as info messages. So using LogLevel should be fine.

File: src/foam/u2/dialog/NotificationMessage.js [hunk 3]
```diff
@@ -20,111 +20,204 @@ foam.CLASS({
...
       var self = this;
 
+      var img;
+      if ( this.type === 'error' ) {
+        img = 'images/round-error-red.svg';
+      } else if ( this.type === 'warning' ) {
+        img = 'images/baseline-warning-yellow.svg';
+      } else if ( this.type === 'info' ) {
+        img = 'images/round-error-blue.svg';
+      } else {
```
Comment (nanoscott) [inline src/foam/u2/dialog/NotificationMessage.js:167]: what is this else supposed to represent? should the default be 'info'? so the fall through should be to images/round-error-blue.svg?

File: src/foam/u2/dialog/NotificationMessage.js [hunk 4]
```diff
@@ -20,111 +20,204 @@ foam.CLASS({
...
+              .attrs({ src: img })
+            .end()
+            .start().addClass(this.myClass('content'))
+              .enableClass(this.myClass('error-content'), this.type === 'error')
+              .enableClass(this.myClass('warning-content'), this.type === 'warning')
+              .enableClass(this.myClass('info-content'), this.type === 'info')
+              .callIfElse(foam.String.isInstance(this.message), function() {
+                this.add(self.message);
+              }, function() {
+                this.tag(self.message);
```
Comment (nanoscott) [inline src/foam/u2/dialog/NotificationMessage.js:180]: add a console.log(self.message); below this tag .. and maybe below the description below as well if it is not empty.

**Comment by jlhughes** [comment]
@blakegreen32 if you could update LogLevel:
1. convert it's custom colour config to the new colour support Kevin added to Enums
2. change the colours themselves to those we've selected for toast notifications.  Use the blue for debug.

File: src/foam/u2/dialog/NotificationMessage.js (hunk 1 above)
Comment (nanoscott) [reply src/foam/u2/dialog/NotificationMessage.js:140]: So we are going to do the LogLevel migration in a follow-up task, not in this PR. We can resolve this comment in this PR and address it later.

**Review by nanoscott — APPROVED** [review]

**Pull request merged**